### DISCUSSION TITLE
Static php-apache2 app redirects to insecure url (http) instead of secure (https)

### DISCUSSION THREAD

**Discussion by alancwoo** [dokku/dokku#7054] [Q&A]
I have an incredibly simple php-apache2 app just to serve some static data from a storage mount.

For some reason, visiting the app at `https://example.org/folder` creates a redirect to `http://example.org/folder/` instead of the secure path.

Visiting `https://example.org/folder/` or `https://example.org/folder/index.html` works fine, as no redirect is created.

Letsencrypt is enabled and running on the app. Also, the site is behind cloudflare with a secure proxy.

Any advice would be greatly appreciated as this becomes a problem when embedding the site in an iframe which fails to load due to an mixed/insecure embed.

The network inspector shows

```
General:
Request URL: https://example.org/folder
Request Method: GET
Status Code: 301 Moved Permanently
Remote Address: REDACTED:443
Referrer Policy: strict-origin-when-cross-origin

Response Headers:
content-length: 268
content-type: text/html; charset=iso-8859-1
date: Wed, 14 Aug 2024 16:18:29 GMT
location: http://example.org/folder/
server: nginx
strict-transport-security: max-age=15724800; includeSubdomains
```

**Comment by josegonzalez** [comment]
Can you check your app logs to see if the request is hitting your server? I don't think our nginx config would ever do this sort of redirect - only http => https.

**Reply by alancwoo** [reply]
Thank you for the response, I am seeing both hits on the app side I believe:

```
2024-08-16T03:34:21.883633915Z app[web.1]: 172.17.0.1 - - [16/Aug/2024:03:34:21 +0000] "GET /folder HTTP/1.1" 301 256 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
2024-08-16T03:34:21.883690357Z app[web.1]: 172.17.0.1 - - [16/Aug/2024:03:34:21 +0000] "GET /folder/ HTTP/1.1" 200 2909 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
```

The app has nothing in it except a Procfile with `web: vendor/bin/heroku-php-apache2` and a storage link. I'm confused where the redirect is coming from.

**Reply by josegonzalez** [reply]
Okay that looks like your app config is normalizing the request on its own, and possibly returning a redirect _without_ the https on it then. If you weren't seeing any requests for the `GET /folder` in your app logs, then I'd expect that to be on the Dokku side.

I would investigate whatever is doing the redirect (maybe its in app, maybe its your apache config) and fixing that.
